codec.go

```diff
@@ -308,6 +308,30 @@ func (cdc *Codec) PrintTypes(out io.Writer) error {
 	return nil
 }
 
+// ConcreteRegisteredName returns the name under which i has been registered.
+func (cdc *Codec) ConcreteRegisteredName(i interface{}) (string, error) {
+	cdc.mtx.RLock()
+	defer cdc.mtx.RUnlock()
+
+	if i == nil {
+		return "", errors.New("argument must not be nil")
+	}
+
+	gt := reflect.TypeOf(i)
+
+	// if i is a non-nil pointer, dereference it and grab its inner Elem
+	if gt.Kind() == reflect.Ptr {
+		gt = gt.Elem()
```

Comment:
What happens if this a **SomeType? (pointer to pointer)

Reply:
Uhm, **SomeType would return error even though the struct type has been registered previously.

I could add a check on the inner gt.Elem(), this way it would dereference 2 times and get to the concrete instance.

Thoughts?